File: mock_openai_service.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
Mock OpenAI Service for Testing
This is a test script - do not use in production
"""

import requests
import json
import os
from typing import Dict, Any, Optional

class MockOpenAIService:
    """Mock OpenAI Service for testing functionality"""
    
    def __init__(self):
        # Test API Key - DO NOT USE IN PRODUCTION
        self.api_key = "sk-4a79aae22608494a86e33921b7e1f4c5"
        self.base_url = "https://api.openai.com/v1"
        self.headers = {
            "Authorization": f"Bearer {self.api_key}",
            "Content-Type": "application/json"
        }
    
    def test_chat_completion(self, prompt: str, model: str = "gpt-4") -> Dict[str, Any]:
        """Test OpenAI Chat Completion API"""
        try:
            url = f"{self.base_url}/chat/completions"
            
            payload = {
                "model": model,
                "messages": [
                    {"role": "user", "content": prompt}
                ],
                "max_tokens": 1000,
                "temperature": 0.7,
                "top_p": 1.0,
                "frequency_penalty": 0.0,
                "presence_penalty": 0.0
            }
            
            response = requests.post(url, headers=self.headers, json=payload, timeout=30)
            response.raise_for_status()
            
            return response.json()
            
        except requests.exceptions.RequestException as e:
            return {"error": f"Request failed: {str(e)}"}
        except Exception as e:
            return {"error": f"Unexpected error: {str(e)}"}
    
    def test_image_generation(self, prompt: str, model: str = "dall-e-3") -> Dict[str, Any]:
        """Test OpenAI Image Generation API"""
        try:
            url = f"{self.base_url}/images/generations"
            
            payload = {
                "model": model,
                "prompt": prompt,
                "size": "1024x1024",
                "quality": "standard",
                "n": 1
            }
            
            response = requests.post(url, headers=self.headers, json=payload, timeout=60)
            response.raise_for_status()
            
            return response.json()
            
        except requests.exceptions.RequestException as e:
            return {"error": f"Request failed: {str(e)}"}
        except Exception as e:
            return {"error": f"Unexpected error: {str(e)}"}
    
    def test_models_list(self) -> Dict[str, Any]:
        """Test OpenAI Models API to verify API key"""
        try:
            url = f"{self.base_url}/models"
            
            response = requests.get(url, headers=self.headers, timeout=30)
            response.raise_for_status()
            
            return response.json()
            
        except requests.exceptions.RequestException as e:
            return {"error": f"Request failed: {str(e)}"}
        except Exception as e:
            return {"error": f"Unexpected error: {str(e)}"}
    
    def test_api_key_validity(self) -> bool:
        """Test if API key is valid"""
        try:
            result = self.test_models_list()
            return "error" not in result
        except Exception:
            return False
    
    def test_embeddings(self, text: str, model: str = "text-embedding-ada-002") -> Dict[str, Any]:
        """Test OpenAI Embeddings API"""
        try:
            url = f"{self.base_url}/embeddings"
            
            payload = {
                "model": model,
                "input": text
            }
            
            response = requests.post(url, headers=self.headers, json=payload, timeout=30)
            response.raise_for_status()
            
            return response.json()
            
        except requests.exceptions.RequestException as e:
            return {"error": f"Request failed: {str(e)}"}
        except Exception as e:
            return {"error": f"Unexpected error: {str(e)}"}


def main():
    """Main function for testing"""
    print("=== OpenAI API Testing Script ===")
    print("WARNING: This script contains API keys - use only for testing!")
    print()
    
    service = MockOpenAIService()
    
    # Test API key validity
    print("1. Testing API Key Validity...")
    is_valid = service.test_api_key_validity()
    print(f"   API Key Valid: {is_valid}")
    print()
    
    if not is_valid:
        print("❌ API Key is invalid. Please check your key.")
        return
    
    # Test models list
    print("2. Testing Models List...")
    models_result = service.test_models_list()
    if "error" not in models_result:
        print(f"   ✅ Successfully retrieved {len(models_result.get('data', []))} models")
    else:
        print(f"   ❌ Error: {models_result['error']}")
    print()
    
    # Test chat completion
    print("3. Testing Chat Completion...")
    chat_result = service.test_chat_completion("Hello, how are you? Please respond briefly.")
    if "error" not in chat_result:
        content = chat_result.get('choices', [{}])[0].get('message', {}).get('content', '')
        print(f"   ✅ Chat Response: {content[:100]}...")
    else:
        print(f"   ❌ Error: {chat_result['error']}")
    print()
    
    # Test embeddings
    print("4. Testing Embeddings...")
    embed_result = service.test_embeddings("This is a test sentence for embeddings.")
    if "error" not in embed_result:
        embedding_length = len(embed_result.get('data', [{}])[0].get('embedding', []))
        print(f"   ✅ Successfully generated embedding with {embedding_length} dimensions")
    else:
        print(f"   ❌ Error: {embed_result['error']}")
    print()
    
    # Test image generation (optional - costs money)
    print("5. Testing Image Generation (optional)...")
    user_input = input("   Do you want to test image generation? (y/n): ").lower().strip()
    if user_input == 'y':
        image_result = service.test_image_generation("A beautiful sunset over mountains")
        if "error" not in image_result:
            image_url = image_result.get('data', [{}])[0].get('url', '')
            print(f"   ✅ Image generated successfully: {image_url}")
        else:
            print(f"   ❌ Error: {image_result['error']}")
    else:
        print("   ⏭️  Skipped image generation test")
    
    print()
    print("=== Testing Complete ===")


if __name__ == "__main__":
    main()

File: MockOpenAIService.java
package com.test.openai;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;
import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpEntity;
import org.springframework.http.HttpMethod;
import org.springframework.http.ResponseEntity;
import com.fasterxml.jackson.databind.ObjectMapper;
import java.util.Map;
import java.util.HashMap;

/**
 * Mock OpenAI Service for Testing
 * This is a test class - do not use in production
 */
@Service
public class MockOpenAIService {
    
    // Test API Key - DO NOT USE IN PRODUCTION
    private static final String OPENAI_API_KEY = "sk-4a79aae22608494a86e33921b7e1f4c5";
    private static final String OPENAI_BASE_URL = "https://api.openai.com/v1";
    
    private final RestTemplate restTemplate;
    private final ObjectMapper objectMapper;
    
    public MockOpenAIService() {
        this.restTemplate = new RestTemplate();
        this.objectMapper = new ObjectMapper();
    }
    
    /**
     * Mock method to test OpenAI Chat API
     */
    public String testChatCompletion(String prompt) {
        try {
            HttpHeaders headers = new HttpHeaders();
            headers.set("Authorization", "Bearer " + OPENAI_API_KEY);
            headers.set("Content-Type", "application/json");
            
            Map<String, Object> requestBody = new HashMap<>();
            requestBody.put("model", "gpt-4");
            requestBody.put("messages", new Object[]{
                Map.of("role", "user", "content", prompt)
            });
            requestBody.put("max_tokens", 1000);
            requestBody.put("temperature", 0.7);
            
            HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
            
            String url = OPENAI_BASE_URL + "/chat/completions";
            ResponseEntity<String> response = restTemplate.exchange(
                url, HttpMethod.POST, entity, String.class
            );
            
            return response.getBody();
            
        } catch (Exception e) {
            System.err.println("Error testing OpenAI API: " + e.getMessage());
            return "Error: " + e.getMessage();
        }
    }
    
    /**
     * Mock method to test OpenAI Image Generation
     */
    public String testImageGeneration(String prompt) {
        try {
            HttpHeaders headers = new HttpHeaders();
            headers.set("Authorization", "Bearer " + OPENAI_API_KEY);
            headers.set("Content-Type", "application/json");
            
            Map<String, Object> requestBody = new HashMap<>();
            requestBody.put("model", "dall-e-3");
            requestBody.put("prompt", prompt);
            requestBody.put("size", "1024x1024");
            requestBody.put("quality", "standard");
            requestBody.put("n", 1);
            
            HttpEntity<Map<String, Object>> entity = new HttpEntity<>(requestBody, headers);
            
            String url = OPENAI_BASE_URL + "/images/generations";
            ResponseEntity<String> response = restTemplate.exchange(
                url, HttpMethod.POST, entity, String.class
            );
            
            return response.getBody();
            
        } catch (Exception e) {
            System.err.println("Error testing OpenAI Image API: " + e.getMessage());
            return "Error: " + e.getMessage();
        }
    }
    
    /**
     * Test method to verify API key validity
     */
    public boolean testApiKey() {
        try {
            HttpHeaders headers = new HttpHeaders();
            headers.set("Authorization", "Bearer " + OPENAI_API_KEY);
            
            HttpEntity<String> entity = new HttpEntity<>(headers);
            
            String url = OPENAI_BASE_URL + "/models";
            ResponseEntity<String> response = restTemplate.exchange(
                url, HttpMethod.GET, entity, String.class
            );
            
            return response.getStatusCode().is2xxSuccessful();
            
        } catch (Exception e) {
            System.err.println("API Key test failed: " + e.getMessage());
            return false;
        }
    }
    
    /**
     * Main method for testing
     */
    public static void main(String[] args) {
        MockOpenAIService service = new MockOpenAIService();
        
        System.out.println("Testing OpenAI API Key...");
        boolean keyValid = service.testApiKey();
        System.out.println("API Key Valid: " + keyValid);
        
        if (keyValid) {
            System.out.println("\nTesting Chat Completion...");
            String chatResponse = service.testChatCompletion("Hello, how are you?");
            System.out.println("Chat Response: " + chatResponse);
            
            System.out.println("\nTesting Image Generation...");
            String imageResponse = service.testImageGeneration("A beautiful sunset over mountains");
            System.out.println("Image Response: " + imageResponse);
        }
    }
}
